jetty-http/src/main/java/org/eclipse/jetty/http/UriCompliance.java

```diff
@@ -62,19 +74,46 @@ public String getName()
         @Override
         public String getURL()
         {
-            return url;
+            return _url;
         }
 
         @Override
         public String getDescription()
         {
-            return description;
+            return _description;
         }
     }
 
-    public static final UriCompliance SAFE = new UriCompliance("SAFE", noneOf(Violation.class));
-    public static final UriCompliance STRICT = new UriCompliance("STRICT", allOf(Violation.class));
-    private static final List<UriCompliance> KNOWN_MODES = Arrays.asList(SAFE, STRICT);
+    /**
+     * The default compliance mode that extends RFC3986 compliance with additional violations to avoid ambiguous URIs
+     */
+    public static final UriCompliance DEFAULT = new UriCompliance("DEFAULT", noneOf(Violation.class));
+
+    /**
+     * LEGACY compliance mode that disallows only ambiguous path parameters as per Jetty-9.4
+     */
+    public static final UriCompliance LEGACY = new UriCompliance("LEGACY", EnumSet.of(Violation.AMBIGUOUS_PATH_SEGMENT, Violation.AMBIGUOUS_PATH_SEPARATOR));
+
+    /**
+     * Compliance mode that exactly follows RFC3986, including allowing all additional ambiguous URI Violations
+     */
+    public static final UriCompliance RFC3986 = new UriCompliance("RFC3986", allOf(Violation.class));
+
+    /**
+     * @deprecated equivalent to DEFAULT
+     */
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    @Deprecated
+    public static final UriCompliance SAFE = new UriCompliance("SAFE", DEFAULT.getAllowed());
+
+    /**
+     * @deprecated equivalent to RFC3986
+     */
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    @Deprecated
+    public static final UriCompliance STRICT = new UriCompliance("STRICT", RFC3986.getAllowed());
+
+    private static final List<UriCompliance> KNOWN_MODES = Arrays.asList(DEFAULT, LEGACY, RFC3986, STRICT, SAFE);
```

Comment:
This is `private` and only referenced by a javadoc, so it cannot even be seen -- I'd say remove it.